### PR TITLE
Migrate Query page to React: Save changes

### PR DESCRIPTION
**Pull request by kravets-levko**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Related Tickets & Documents

getredash/redash#4429

### CODE REVIEW

File: client/app/pages/queries/QuerySource.jsx [hunk 1]
```diff
@@ -41,6 +42,98 @@ function chooseDataSourceId(dataSourceIds, availableDataSources) {
...
+      "description",
+      "name",
+      "data_source_id",
+      "options",
+      "latest_query_data_id",
+      "is_draft",
+    ]);
+  }
+
+  // omit pendingValue before saving
```
Comment (gabrieldutra) [inline client/app/pages/queries/QuerySource.jsx]:
```suggestion
  // omit parameter properties that don't need to be stored
```
(or whatever like this, I forgot to update this comment in the Parameter structure update PR :sweat_smile:)

File: client/app/pages/queries/QuerySource.jsx [hunk 2]
```diff
@@ -41,6 +42,98 @@ function chooseDataSourceId(dataSourceIds, availableDataSources) {
...
+      onCancel: () => {
+        reject();
+      },
+      maskClosable: true,
+      autoFocusButton: null,
+    });
+  });
+}
+
+function saveQuery(query, data, options) {
```
Comment (gabrieldutra) [inline client/app/pages/queries/QuerySource.jsx]: These methods will be shared with the QueryView, so perhaps have them in a separate file?

Comment (kravets-levko) [reply client/app/pages/queries/QuerySource.jsx]: Of course :slightly_smiling_face: Already doing that :+1:

Comment (kravets-levko) [reply client/app/pages/queries/QuerySource.jsx]: Done

File: client/app/pages/queries/QuerySource.jsx (hunk 1 above)
Comment (kravets-levko) [reply client/app/pages/queries/QuerySource.jsx]: Done :+1:

**Comment by gabrieldutra** [comment]
@kravets-levko is there ath else you want to do in here, or :rocket:?

**Comment by kravets-levko** [comment]
@gabrieldutra 

> is there ath else you want to do in here

Yes - implementing `saveQuery` function was a half of job; another half is to use it :slightly_smiling_face:  Also, I implemented fork/archive query actions. I think we could proceed with this :+1:

File: client/app/pages/queries/utils/archiveQuery.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      autoFocusButton: null,
+    });
+  });
+}
+
+function doArchiveQuery(query) {
+  // Prettier will put `.$promise` before `.catch` on next line :facepalm:
+  // prettier-ignore
+  return Query.delete({ id: query.id }).$promise
+    .then(() => Promise.resolve(extend(clone(query), { is_archived: true, schedule: null })))
```
Comment (gabrieldutra) [inline client/app/pages/queries/utils/archiveQuery.jsx]: I'd have preferred `Promise.resolve({ ...query, is_archived: true, schedule: null })`, but that works

Comment (kravets-levko) [reply client/app/pages/queries/utils/archiveQuery.jsx]: Spread operator will create a plain object and copy properties to it - so query will loose it's prototype with all methods (like `.isNew()`). Therefore I used `clone` + `extend` here

**Pull request merged**